components/config-mgmt-service/integration_test/node_export_test.go

```diff
@@ -0,0 +1,498 @@
+		},
+		{
+			description: "One node with several projects where one matches one of several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+			},
+			allowedProjects:   []string{"project3", "project10", "project12", "project13"},
+			expectedNodeNames: []string{"1"},
+		},
+		{
+			description: "Two nodes with several projects where one matches one of several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "2",
+					},
+					Projects: []string{"project13", "project14", "project17", "project16"},
+				},
+			},
+			allowedProjects:   []string{"project3", "project10", "project12", "project13"},
+			expectedNodeNames: []string{"1", "2"},
+		},
+		{
+			description: "One node with several projects where none matches several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+			},
+			allowedProjects:   []string{"project14", "project10", "project12", "project13"},
+			expectedNodeNames: []string{},
+		},
+		{
+			description: "One node with several projects where two matches two of several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project10", "project7", "project6"},
+				},
+			},
+			allowedProjects:   []string{"project3", "project10", "project12", "project13"},
+			expectedNodeNames: []string{"1"},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(fmt.Sprintf("Project filter: %s", test.description), func(t *testing.T) {
+			// Adding required node data
+			for index := range test.nodes {
+				test.nodes[index].Exists = true
+				test.nodes[index].NodeInfo.EntityUuid = newUUID()
+			}
+
+			// Add nodes with projects
+			suite.IngestNodes(test.nodes)
+			defer suite.DeleteAllDocuments()
+			ctx := auth_context.NewOutgoingContext(contextWithProjects(test.allowedProjects))
+			response, err := client.NodeExport(ctx, &request.NodeExport{
+				OutputType: "json",
+				Filter:     test.filter,
+			})
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			data := make([]byte, 0)
+			for {
+				tdata, err := response.Recv()
+				if err != nil && err == io.EOF {
+					data = append(data, tdata.GetContent()...)
+					break
+				}
+
+				require.NoError(t, err)
+				data = append(data, tdata.GetContent()...)
+			}
+
+			actualNodeNames := make([]string, 0)
+			jsonparser.ArrayEach(data, func(node []byte, _ jsonparser.ValueType, _ int, err error) {
+				if err == nil {
```

Comment:
I know this test is not trying to verify basic correctness of the endpoint, but it feels bad to ignore errors here...

Reply:
That sounds right. I will update this test to fail on a returned error.